src/Concordium/Client/Types/Contract/Parameter.hs

```diff
@@ -96,8 +100,45 @@ getJSONUsingSchema typ = case typ of
       Just idx -> let (contractName, funcNameWithDot) = Text.splitAt idx rawName
                       funcName = Text.tail funcNameWithDot -- Safe, since we know it contains a dot.
                   in return $ AE.object ["contract" .= contractName, "func" .= funcName]
+  ULeb128 constraint -> AE.toJSON . show <$> leb128 0
+    where
+      leb128 :: Int -> S.Get Integer
+      leb128 byteCount = do
+        unless (byteCount < fromIntegral constraint) $ fail "Leb128 encoding violated the constraint on the number of bytes"
+        byte <- S.getWord8
+        let value = fromIntegral (Bits.clearBit byte 7)
+        let moreBytes = Bits.testBit byte 7
+        if moreBytes
+          then do
+            next <- leb128 (byteCount + 1)
+            return $ value + 128 * next
+          else return value
+  ILeb128 constraint -> do
+    AE.toJSON . show <$> ileb128 0 0
+    where
+      ileb128 :: Int -> Integer -> S.Get Integer
+      ileb128 byteCount acc = do
+        unless (byteCount < fromIntegral constraint) $ fail "Leb128 encoding violated the constraint on the number of bytes"
+        byte <- S.getWord8
+        let byteValue = fromIntegral (Bits.clearBit byte 7)
+        let value = acc + (2 ^ (7 * byteCount)) * byteValue
+        let moreBytes = Bits.testBit byte 7
+        if moreBytes
+          then do
+            ileb128 (byteCount + 1) value
+          else let isNegative = Bits.testBit byte 6
+               in return $ if isNegative
+                             then value - 2 ^ ((byteCount + 1) * 7)
+                             else value
+  ByteList sl -> do
+    byteList <- getListOfWithSizeLen sl S.getWord8
+    return $ AE.toJSON $ toHex $ BS.pack byteList
+  ByteArray len -> do
+    bytes <- S.getByteString (fromIntegral len)
+    return $ AE.toJSON $ toHex bytes
 
   where
+    toHex = Text.decodeUtf8 . BS16.encode
```

Comment:
`Text.decodeUtf8` is a partial function and will throw an error if the input is not valid UTF8.
I am not sure whether that would always be the case here. Please add a comment about why it is safe, or use the safe version `Text.decodeUtf8'`.

Reply:
I have added a comment